### PR TITLE
deprecate the rest of DSA support for SSH

### PR DESCRIPTION
**Pull request by reaperhulk**

### CODE REVIEW

File: src/cryptography/hazmat/backends/openssl/backend.py [hunk 1]
```diff
@@ -1616,6 +1616,13 @@ def _private_key_bytes(
...
         if format is serialization.PrivateFormat.OpenSSH:
             if encoding is serialization.Encoding.PEM:
+                if isinstance(key, dsa.DSAPrivateKey):
+                    warnings.warn(
+                        "SSH DSA key support is deprecated and will be "
+                        "removed in a future release",
+                        utils.DeprecatedIn40,
+                        stacklevel=3,
+                    )
                 return ssh._serialize_ssh_private_key(
```
Comment (alex) [inline src/cryptography/hazmat/backends/openssl/backend.py:1626]: Can we put these on the `ssh` module side for consistency?

File: src/cryptography/hazmat/backends/openssl/backend.py [hunk 2]
```diff
@@ -1697,6 +1704,13 @@ def _public_key_bytes(
...
         if format is serialization.PublicFormat.OpenSSH:
             if encoding is serialization.Encoding.OpenSSH:
+                if isinstance(key, dsa.DSAPublicKey):
+                    warnings.warn(
+                        "SSH DSA key support is deprecated and will be "
+                        "removed in a future release",
+                        utils.DeprecatedIn40,
+                        stacklevel=3,
+                    )
                 return ssh.serialize_ssh_public_key(key)
```
Comment (alex) [inline src/cryptography/hazmat/backends/openssl/backend.py:1714]: Why doesn't this have a leading underscore :-/

File: tests/hazmat/primitives/test_ssh.py [hunk 3]
```diff
@@ -153,63 +178,66 @@ def test_load_ssh_private_key(self, key_file, backend):
...
-                Encoding.OpenSSH, PublicFormat.OpenSSH
-            )
-            == nocomment_data
-        )
+        for data in (
+            priv_data,
+            bytearray(priv_data),
+            memoryview(priv_data),
+            memoryview(bytearray(priv_data)),
+        ):
```
Comment (alex) [inline tests/hazmat/primitives/test_ssh.py:186]:
```suggestion
        for data in [
            priv_data,
            bytearray(priv_data),
            memoryview(priv_data),
            memoryview(bytearray(priv_data)),
        ]:
```

File: src/cryptography/hazmat/backends/openssl/backend.py (hunk 2 above)
Comment (reaperhulk) [reply src/cryptography/hazmat/backends/openssl/backend.py:1714]: Yeah, not great. We never documented it but named it wrong.

**Review by alex — APPROVED** [review]

**Pull request merged**